src/main/java/com/odysseusinc/arachne/datanode/controller/ExceptionHandlingAdvice.java

```diff
@@ -100,21 +102,29 @@ public ResponseEntity handleDataAccessExceptions(Exception ex) {
         return  getErrorResponse(result,  ex);
     }
 
-    private ResponseEntity<JsonResult> getErrorResponse(JsonResult result, Exception ex) {
+    private ResponseEntity<JsonResult> getErrorResponse(final JsonResult result, final Exception ex) {
 
-        result.setErrorMessage(ex.getMessage());
+        final String message = getErrorMessage(result, ex);
+        
+        result.setErrorMessage(message);
 
         if (errorsTokenEnabled) {
-            String errorToken = generateErrorToken();
-            LOGGER.error(ex.getMessage() + " token: " + errorToken, ex);
+            final String errorToken = generateErrorToken();
+            LOGGER.error(message + " token: " + errorToken, ex);
             result.setErrorMessage(String.format(ERROR_MESSAGE_WITH_TOKEN, errorToken));
+        } else {
+            LOGGER.error(message, ex);
         }
-
-        LOGGER.error(ex.getMessage(), ex);
-        result.setErrorMessage(ERROR_MESSAGE);
+        
         return new ResponseEntity<>(result, HttpStatus.OK);
     }
 
+    private String getErrorMessage(final JsonResult result, final Exception ex) {
```

Comment:
Did you check case with IntegrationValidationException?

Reply:
In case of this exception it works properly

Reply:
ok, thanks